images/operator-sdk/kustomize.Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+
+RUN microdnf install -y golang make which
+RUN microdnf install -y git
+
+ARG BIN=operator-sdk
+COPY $BIN /usr/local/bin/operator-sdk
+
+# install kustomize
+RUN git clone https://github.com/kubernetes-sigs/kustomize.git
+RUN cd kustomize && \
+    cd kustomize && \
+    go install .
+RUN ~/go/bin/kustomize version
```

Comment:
A few suggestions:
* Can you move the kustomize installation to the existing `operator-sdk` [Dockerfile](https://github.com/operator-framework/operator-sdk/blob/master/images/operator-sdk/Dockerfile)? I don't think we need separate images.
* Do pre-built `kustomize` binaries exist for all the OS/ARCH pairs we care about? If so, we should default to directly downloading them. If not, we should use a multi-stage Dockerfile to build the binaries in a build stage and copy them into the final image. We shouldn't add unnecessary source code and bloat to the final image.
* we need to use the version of `kustomize` that is currently supported by operator-sdk. That version is v3.5.4. I would suggest we pass that version in via a Dockerfile ARG so that it's easier for us to centrally manage.
* If we do need to build from source, use `go get` instead of `git clone && go install`

Reply:
Same for `controller-gen`. Kindly consider adding `controller-gen` to the existing base `opeartor-sdk` [Dockerfile](https://github.com/operator-framework/operator-sdk/blob/master/images/operator-sdk/Dockerfile) as requested in the original issue #4000

Reply:
@joelanford I do not believe kustomize has builds for all of the OS/ARCH pairs. I can create a multi-stage Dockerfile to build the binaries.